nni/algorithms/compression/pytorch/pruning/net_adapt_pruner.py

```diff
@@ -84,31 +84,34 @@ def evaluator(model):
 
     def __init__(self, model, config_list, short_term_fine_tuner, evaluator,
                  optimize_mode='maximize', base_algo='l1', sparsity_per_iteration=0.05, experiment_data_dir='./'):
-        # models used for iterative pruning and evaluation
-        self._model_to_prune = copy.deepcopy(model)
         self._base_algo = base_algo
-
-        super().__init__(model, config_list)
-
         self._short_term_fine_tuner = short_term_fine_tuner
         self._evaluator = evaluator
         self._optimize_mode = OptimizeMode(optimize_mode)
 
         # hyper parameters for NetAdapt algorithm
         self._sparsity_per_iteration = sparsity_per_iteration
 
-        # overall pruning rate
-        self._sparsity = config_list[0]['sparsity']
-
-        # config_list
-        self._config_list_generated = []
-
         self._experiment_data_dir = experiment_data_dir
         if not os.path.exists(self._experiment_data_dir):
             os.makedirs(self._experiment_data_dir)
 
         self._tmp_model_path = os.path.join(self._experiment_data_dir, 'tmp_model.pth')
 
+        super().__init__(model, config_list)
+
+    def reconfig(self, model, config_list):
+        # models used for iterative pruning and evaluation
+        self._model_to_prune = copy.deepcopy(model)
```

Comment:
if the model is pruned from the original state just with a different sparsity ratio, what is the benefit? in other words, what type of information is inherited between two consecutive iteration?

Reply:
It is a good question and in `net_adapt`, it will find just a layer that has the best performance for the current iteration sparsity. Then update config_list.
i.e. iteratoin one, `sparsity=0.1` `conv1 performance: 0.8; conv2 performance: 0.6`, then `[] <- {'op_names': 'conv1', 'sparsity': 0.1}`.
iteratoin two, `sparsity=0.2`  `conv1 performance: 0.4; conv2 performance: 0.7`, then `[{'op_names': 'conv1', 'sparsity': 0.1}] <- {'op_names': 'conv2', 'sparsity': 0.2}`.
...